### PR TITLE
Explicitly set GITHUB_TOKEN permissions for yocto workflow

### PR DESCRIPTION
**Pull request by rcooke-warwick**

### CODE REVIEW

**Review by klutchell — APPROVED** [review]

File: .github/workflows/iot-gate-imx8plus-d1d8.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ permissions:
 jobs:
   yocto:
     name: Yocto
-    uses: balena-os/balena-yocto-scripts/.github/workflows/yocto-build-deploy.yml@f170ea86c0476f46e0144528bcc1387c9da07e11 # v1.27.14
```
Comment (floion) [inline .github/workflows/iot-gate-imx8plus-d1d8.yml:38]: why is this needed?
